components/sendgrid/actions/send-email-single-recipient/send-email-single-recipient.mjs

```diff
@@ -266,6 +278,45 @@ export default {
         replyTo.name = this.replyToName;
       }
     }
+
+    //Prepares the asm object or the asmGroupId. Sendgrid API requires the params to be
+    //an int, so make sure it can be parsed as one
+    let asmConfig = undefined;
+    if (this.asm) {
+      if (this.asm.group_id) {
+        const groupId = parseInt(this.asm.group_id, 10);
+        if (!isNaN(groupId)) {
+          asmConfig = {
+            group_id: groupId,
+          };
+        }
+      }
+      //If the asmGroupId param was set, configure that
+      if (this.asmGroupId) {
+        asmConfig = {
+          group_id: this.asmGroupId,
+        };
+      }
+      //If asm.groups_to_display is configured, parse that and copy over
+      if (this.asm?.groups_to_display) {
+        const groups = JSON.parse(this.asm.groups_to_display);
+        const groupIds = [];
+        for (let i = 0; i < groups.length; i++) {
+          const groupId = parseInt(groups[i], 10);
+          if (!isNaN(groupId)) {
+            groupIds.push(groupId);
+          }
+        }
+        if (groupIds.length > 0) {
+          asmConfig.groups_to_display = groupIds;
+        }
+      }
+      //if asmGroupsToDisplay is configured, copy that over
+      if (this.asmGroupsToDisplay) {
+        asmConfig.groups_to_display = this.asmGroupsToDisplay;
+      }
+    }
```

Comment:
This entire block of code is the same for both actions - perhaps it should be a method that is common to both of them?

e.g. a `getAsmConfig` method with this code can be defined in a common file that is imported and called by both components:
```
const asmConfig = this.getAsmConfig(); // returns undefined if none of the conditions are met
```

Reply:
Makes sense; I'll move it to the sendgrid object.